internal/jwtparser/jwt.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package jwtparser
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/juju/errors"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+// Parser is a JWT parser responsible for parsing JWT tokens.
+type Parser struct {
+	cache      *jwk.Cache
+	httpClient jwk.HTTPClient
+	refreshURL string
```

Comment:
Because this parser is being shared across go routine's I think we should make it thread safe now as there isn't any other mechanism in place to stop bad callers.

All we need to do is place a RWMutex around the refreshURL.